### PR TITLE
daemon, cli: remove jobs when remove pipeline

### PR DESCRIPTION
**Pull request by FeelyChau**
```sh
$ pipcook pipeline remove 42lw3pir   
? 1 job which belong to the pipeline will be removed too, continue? Yes
ℹ 1 jobs removed.
✔ pipeline 42lw3pir has removed.
$ pipcook pipeline remove all
? 1 job which belong to the pipeline will be removed too, continue? Yes
ℹ 1 jobs removed.
✔ all pipelines removed.
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -107,6 +107,9 @@ jobs:
       run: |
         ./packages/cli/dist/bin/pipcook run https://raw.githubusercontent.com/alibaba/pipcook/master/example/pipelines/text-bayes-classification.json
         ls ./output && rm -rf ./output
+    - name: remove pipelines
+      run: |
+        ./packages/cli/dist/bin/pipcook remove all
```
Comment (yorkie) [inline .github/workflows/build.yml:112]: Should be pipcook pipeline remove all?

File: packages/cli/src/service/pipeline.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import * as Url from 'url';
 import { PipelineResp, PluginStatusValue, PipelineConfig } from '@pipcook/sdk';
 import { constants, PluginStatus } from '@pipcook/pipcook-core';
 import { readJson } from 'fs-extra';
+import { prompt } from 'inquirer';
```
Comment (yorkie) [inline packages/cli/src/service/pipeline.ts:6]: I think it's not good behavior to write cli logic in service.

File: .github/workflows/build.yml (hunk 1 above)
Comment (FeelyChau) [reply .github/workflows/build.yml]: Fixed.

File: packages/cli/src/service/pipeline.ts (hunk 2 above)
Comment (FeelyChau) [reply packages/cli/src/service/pipeline.ts]: Updated.

File: packages/cli/src/bin/pipcook-pipeline.ts [hunk 3]
```diff
@@ -41,9 +42,30 @@ program
...
+          type: 'confirm',
+          name: 'remove',
+          message: `${jobs.length} ${jobs.length > 1 ? 'jobs' : 'job'} which belong to the pipeline will be removed too, continue?`,
+          default: false
+        }
+      ]);
+      confirm = answer.remove;
+    }
+    if (confirm) {
+      remove(id, jobs, opts);
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook-pipeline.ts:66]:
```suggestion
      await remove(id, jobs, opts);
```

File: packages/daemon/src/app/controller/pipeline.ts [hunk 4]
```diff
@@ -77,16 +77,24 @@ export class PipelineController extends BaseEventController {
...
   /**
-   * delete pipeline by id
+   * delete pipeline by id, it will remove the jobs which belong to the pipeline.
    */
   @del('/:id')
   public async removeOne() {
     const { id } = this.ctx.params;
+    const jobs = await this.pipelineService.getJobsByPipelineId(id);
+    for (const job of jobs) {
+      await this.pipelineService.removeJobByModel(job);
```
Comment (yorkie) [inline packages/daemon/src/app/controller/pipeline.ts:96]: Shall we remove them in parallel?

Comment (FeelyChau) [reply packages/daemon/src/app/controller/pipeline.ts:96]: Updated.

**Review by yorkie — APPROVED** [review]

**Review by yorkie — APPROVED** [review]

**Pull request merged**